File: ApnaCollege/Arrays_II/sort-colors.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zeros, ones, twos = 0, 0, 0
        for num in nums:
            match num:
                case 0:
                    zeros += 1
                case 1:
                    ones += 1
                case 2:
                    twos += 1
        for i in range(len(nums)):
            if i < zeros:
                nums[i] = 0
            elif i < zeros + ones:
                nums[i] = 1
            else:
                nums[i] = 2

File: ApnaCollege/Arrays_II/SortColors.java
package ApnaCollege.Arrays_II;

class SortColors {
    public void sortColors(int[] nums) {
        int zeros = 0, ones = 0;
        for (int n: nums){
            switch (n) {
                case 0 -> zeros++;
                case 1 -> ones++;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if(i < zeros) {
                nums[i] = 0;
            } else if (i < zeros + ones) {
                nums[i] = 1;
            } else {
                nums[i] = 2;
            }
        }
    }
}